### PR TITLE
Make requirements and setup.py compatible with ginkgo.

### PR DESCRIPTION
**Pull request by jcdyer**
**Description:** Update dependencies to support older versions required by ginkgo

**JIRA:** OC-4474

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

This is a support ticket for edx-solutions/edx-platform#1078.  If that works, this works.  The goal is to keep this compatible with mainline edx-platform as well, so we don't have to create a long-running branch to support ginkgo.

**Reviewers:**
- [ ] @afzaledx 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** None

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,8 @@ matrix:
       env: TOXENV=docs
     - python: 3.5
       env: TOXENV=django20
+    - python: 2.7
+      env: TOXENV=django18
```
Comment (jcdyer) [inline .travis.yml]: edx-solutions runs on django 1.8, so I want to make sure we're testing it.

File: completion_aggregator/signals.py [hunk 2]
```diff
@@ -25,22 +25,35 @@ def register():
...
-        ENROLLMENT_TRACK_UPDATED.connect(cohort_updated_handler)
+
+    ginkgo_error_template = "%s signal not found. If this is a solutions/ginkgo server, this is expected."
+
+    try:
+        from openedx.core.djangoapps.course_groups.signals.signals import COHORT_MEMBERSHIP_UPDATED
+    except ImportError:
+        log.info(ginkgo_error_template, "COHORT_MEMBERSHIP_UPDATED")
+    else:
         COHORT_MEMBERSHIP_UPDATED.connect(cohort_updated_handler)
```
Comment (jcdyer) [inline completion_aggregator/signals.py]: These signals were added after Ginkgo, and I believe represent functionality not used by McKinsey.  @bradenmacdonald can you confirm that McKinsey doesn't need cohorts or enrollment tracks?

File: completion_aggregator/tasks.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def update_aggregators(username, course_key, block_keys=(), force=False):
     """
     user = User.objects.get(username=username)
     course_key = CourseKey.from_string(course_key)
-    block_keys = set(UsageKey.from_string(key) for key in block_keys)
+    block_keys = set(UsageKey.from_string(key).map_into_course(course_key) for key in block_keys)
```
Comment (jcdyer) [inline completion_aggregator/tasks.py]: map_into_course adds the run information to the block keys, which allows them to match the data coming from the block structures.

File: requirements/base.in [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # Core requirements for using this application
 
-celery>=3.0,<4.0          # Asynchronous tasks
-edx-celeryutils           # Custom task extensions
-Django>=1.11              # Web application framework
+celery==3.1.18            # Asynchronous tasks
+edx-celeryutils<0.2      # Custom task extensions
+Django>=1.8              # Web application framework
```
Comment (jcdyer) [inline requirements/base.in]: These versions are required to make this work with ginkgo.

Comment (jcdyer) [reply requirements/base.in]: The requirements/*.txt changes below are the result of running `make upgrade` on the new requirements versions.

File: completion_aggregator/signals.py (hunk 2 above)
Comment (bradenmacdonald) [reply completion_aggregator/signals.py]: @jcdyer Yes, as far as I know, they don't use cohorts nor different enrollment tracks.

**Pull request merged**